src/dimension.c

```diff
@@ -909,10 +922,25 @@ dimension_validate_info(DimensionInfo *info)
 	else
 	{
 		/* Open ("time") dimension */
+		Oid			dimtype = info->coltype;
+
 		info->type = DIMENSION_TYPE_OPEN;
 		info->set_not_null = !not_null_is_set;
+
+		if (OidIsValid(info->partitioning_func))
+		{
+			if (!partitioning_func_is_valid(info->partitioning_func, info->type, info->coltype))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_FUNCTION_DEFINITION),
+						 errmsg("invalid partitioning function"),
```

Comment:
The `create_hypertable` in the SQL defines the open dimension function as a `time_func` and the close as  a `partition_func` we should use the same terminology here and there. Perhaps "Invalid time function"?

Reply:
Yeah, though about this. But there are a couple of issues to consider. 

In `add_dimension`, you only add a single dimension while `create_hypertable` can add two at once. So, there needs to be some way to delineate the two dimensions in the latter case; hence, one called `time_func`.

Reply:
This is about both. At minimum I'd like the terminology to be the same in both places, from my own reading of the code I found it difficult to tell which partitioning function was being worked with. At a higher level, I'm a little uncomfortable with calling this a "time partitioning function" because, with our current terminology around space partitioning, I'd expect a time partitioning function to be analogous, i.e. we'd subdivide the time interval for a chunk into multiple tables based on the hash of the time value (to improve write perf?), fwiw while reviewing, it took me a while to realize this PR wasn't that.

Reply:
That's actually a misunderstanding of what the partitioning_func does (and you can debate the naming, but it is legacy). The function is essentially just  a hash function that  maps a value onto a key space, it knows nothing about the partitioning of that key space. Thus, the semantics are the same in both the time and space case. 

Whether `partitioning_func` is an appropriate name or not is an orthogonal question, which is not related specifically to this PR.

Reply:
I see what you're saying and you're right, the debate over `partitioning_func ` does not belong in this PR. (Would it make sense to add a `time_func` argument to `add_dimension` so that it can be called in the same manner as `create_hypertable` of would that just confuse things more?)

Reply:
I am not sure it is worth adding additional arguments to `add_dimension`. The reason we have two function args in `create_hypertable` is that is supports adding two dimensions in a single call, and obviously those function parameter names must be different. I am not sure `add_dimension` should be modified to adapt to `create_hypertable` since the latter mostly supports adding two dimensions as a convenience